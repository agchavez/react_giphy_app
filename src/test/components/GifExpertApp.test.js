import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from 'enzyme';
import { GifExpertApp } from '../../GifExpertApp';


describe('Pruebas de con <GifExpertApp/>', () => {
    test('Debe mostrarse el componente completo', () => {
        const wrapper = shallow(<GifExpertApp/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe mostrar una lista de categorias', () => {
        const categories = ['avengers', 'league of legends'];
        const wrapper = shallow(<GifExpertApp defaultCategories = {categories}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    
         
    })
    
    
})
