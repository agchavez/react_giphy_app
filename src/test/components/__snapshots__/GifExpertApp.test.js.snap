// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de con <GifExpertApp/> Debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="avengers"
      key="avengers"
    />
    <GifGrid
      category="league of legends"
      key="league of legends"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas de con <GifExpertApp/> Debe mostrarse el componente completo 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
